smolyay/basis.py

```diff
@@ -328,141 +278,138 @@ def derivative(self, x):
-        self._basis_functions = basis_functions
-        self._points = points
-        if len(basis_functions) != len(points):
-            raise IndexError("basis_functions and points must have the "
-                    "same number of elements.")
+    def __init__(self, n):
+        super().__init__()
+        self.n = n
 
     @property
-    def points(self):
-        """list: Sampling points."""
-        return self._points
+    def domain(self):
+        """list: Domain the sample points come from."""
+        return [0, 2 * numpy.pi]
 
     @property
-    def basis_functions(self):
-        """list: Basis functions."""
-        return self._basis_functions
+    def sigma(self):
+        """float: Frequency of the complex exponential."""
+        if self.n % 2 == 1:
+            return (1 + self.n) / 2
+        else:
+            return -self.n / 2
 
+    def _function(self, x):
+        r"""Evaluate the basis function.
 
-class NestedBasisFunctionSet(BasisFunctionSet):
-    """Nested set of basis functions and points.
+        The Trigonometric polynomial is evaluated using the following formula:
 
-    Nested points/basis function grow in levels, such that an approximation
-    of a given level uses not only its sampling points but also all the points
-    at lower levels. Nested sets (similarly to ogres) are like onions.
+        .. math::
 
-    Parameters
-    ---------
-    levels : list of lists
-       Assignment of points/basis functions to each level.
-    """
+            \phi_n(x) = \exp(x i \sigma(n))
 
-    def __init__(self,points,basis_functions,levels):
-        super().__init__(points,basis_functions)
-        self._levels = levels
+        where *n* is the order of trigonometric polynomial and
+        is a non-negative integer, and `sigma` the frequency of the
+        complex exponential
+
+        Parameters
+        ----------
+        x : float
+            One-dimensional point on :math:`[0, 2\pi]`.
+
+        Returns
+        -------
+        float
+            Value of Trigonometric polynomial.
+
+        Raises
+        ------
+        ValueError
+            If input is outside the domain `[0, 2\pi]`
+        """
+        return numpy.exp(x * self.sigma * 1j)
+
+    def _derivative(self, x):
+        r"""Evaluate the derivetive of the trigonometric polynomials.
+
+        Parameters
+        ----------
+        x : float
+            One-dimensional point on :math:`[0, 2\pi]`.
+
+        Returns
+        -------
+        float
+            Value of the derivative of Trigonometric polynomial.
 
-    @property
-    def levels(self):
-        """list: List of lists of indexes for points/functions at each level.
         Raises
         ------
-        IndexError
-            max index must be less than total number of points.
+        ValueError
+            If input is outside the domain `[0, 2\pi]`
         """
-        return self._levels
+        x = numpy.asarray(x)
```

Comment:
Why is this needed here, but not in `_function`? I would think both need it?

Reply:
That is correct. I'll add it.